pydantic_apply/_compat.py

```diff
@@ -76,3 +83,15 @@ def model_copy(self, **kwargs: Any) -> pydantic.BaseModel:
 
         def model_dump(self, **kwargs: Any) -> dict[str, Any]:
             return self.obj.model_dump(**kwargs)
+
+        @contextmanager
+        def disable_setattr_handler_cache(self) -> Generator[None, Any, Any]:
+            old_setattr_handlers = {}
+            if Version(PYDANTIC_VERSION) >= Version("2.11"):
+                old_setattr_handlers = self.obj.__class__.__pydantic_setattr_handlers__
+                self.obj.__class__.__pydantic_setattr_handlers__ = {}
+
+            yield
+
+            if Version(PYDANTIC_VERSION) >= Version("2.11"):
+                self.obj.__class__.__pydantic_setattr_handlers__ = old_setattr_handlers
```

Comment:
This should use `finally` to ensure the reset even happens when there is some exception.

```suggestion
            try:
                yield

            finally:
                if Version(PYDANTIC_VERSION) >= Version("2.11"):
                    self.obj.__class__.__pydantic_setattr_handlers__ = old_setattr_handlers
```

Reply:
Using finally in the context manager 👍